src/index.ts

```diff
@@ -90,6 +97,20 @@ type PipedPreContext<TLeft, TRight> = TLeft extends HasInitPreContext<any, any>
   ? { initPreContext: TRight['initPreContext'] }
   : {};
 
```

Comment:
looks a little different than the others...
does this mean I didn't complete factoring in the other ones?
- [ ] explain

Reply:
no it's looks a little different because of sanitizers return type
it's just different
in all sanitizers our return type for left has sanitizer and right has sanitizer is HasSomethig<TLeftContextIn, TRightContextOut>

Reply:
aha you're right - but I'd say let's add a todo to replace SanitizeParamsContextIn<TLeft> with Parameters<TLeft['sanitizeParams']>[0] and SanitizeParamsContextOut<TRight> with ReturnType<T['sanitizeParams']> ... it has those redundant conditionals, but I'd feel more comfortable chopping it down later
- [ ] add todo to simplify type

Reply:
done